src/main/kotlin/com/sourcegraph/cody/agent/protocol/Position.kt

```diff
@@ -1,20 +1,43 @@
 package com.sourcegraph.cody.agent.protocol
 
 import com.intellij.openapi.editor.Document
+import com.intellij.openapi.editor.LogicalPosition
+import kotlin.math.max
+import kotlin.math.min
 
-data class Position(val line: Int, val character: Int) {
+data class Position(@JvmField val line: Long, @JvmField val character: Long) {
+  constructor(line: Int, character: Int) : this(line.toLong(), character.toLong())
```

Comment:
hmmm, why don't we use a constructor that takes Longs instead? 🤔

Reply:
Because we often create it from IJ which uses ints, and I do not want to add `toLong` in 50 places ;)

Reply:
I was afraid that a proper long value could be accidentally converted to int - if that's not a case we are good

Reply:
Kotlin won't accept such conversion without being explicit :)